node/hyperbahn/service_proxy.js

```diff
@@ -166,7 +173,17 @@ function createServiceChannel(serviceName) {
         }
     }
 
-    svcchan.handler = new RelayHandler(svcchan);
+    var handler = new RelayHandler(svcchan);
+
+    // Decorate a circuit health monitor to egress request handlers.
+    if (mode === 'exit' && self.circuits) {
+        handler = new ServiceHealthProxy({
```

Comment:
Does not look like this approach would avoid any closures. Using a proxy allows us to conditionally inject the circuit breaker, so we can use RelayHandler for ingress and RelayHandler+ServiceProxy for egress and keeps them loosely coupled. ServiceHealthProxy is very thin.